### PR TITLE
[test]Add the string testcase of delete

### PR DESCRIPTION
**Pull request by jingkl**
Signed-off-by: jingkl <jingjing.jia@zilliz.com>
issue: #16833

### CODE REVIEW

File: tests/python_client/testcases/test_delete.py [hunk 1]
```diff
@@ -1064,3 +1069,593 @@ def test_delete_sealed_only(self):
...
+        target: test delete entities twice with string expr
+        method: delete with same expr twice
+        expected: No exception for second deletion
+        """
+        # init collection with nb default data
+        collection_w = self.init_collection_general(prefix, nb=tmp_nb, insert_data=True, primary_field=ct.default_string_field_name)[0]
+
+        # assert delete successfully and no exception
+        collection_w.delete(expr=default_string_expr)
+        collection_w.num_entities
```
Comment (binbinlv) [inline tests/python_client/testcases/test_delete.py:1089]: So no comparison is needed here after deletion? The function of calling "num_entities" is only flush?

File: tests/python_client/testcases/test_delete.py [hunk 2]
```diff
@@ -1064,3 +1069,593 @@ def test_delete_sealed_only(self):
...
+        collection_w.delete(expr=default_string_expr)
+
+    @pytest.mark.xfail(reason="https://github.com/milvus-io/milvus/issues/17067")
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_delete_all_index_with_string(self):
+        """
+        target: test delete after creating index
+        method: 1.create collection , insert data, primary_field is string field
+                2.create string and float index ,delete entities, query 
+                4.search
```
Comment (binbinlv) [inline tests/python_client/testcases/test_delete.py:1100]: Here 4 should be 3

File: tests/python_client/testcases/test_delete.py [hunk 3]
```diff
@@ -1064,3 +1068,579 @@ def test_delete_sealed_only(self):
...
+        assert res_before[0].ids == res_travel[0].ids
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_delete_insert_multi_with_string(self):
+        """
+        target: test delete after multi insert with string
+        method: 1.create a collection with string field is primary
+                2.insert multi times, no flush
+                3.load
+                3.delete even number
```
Comment (binbinlv) [inline tests/python_client/testcases/test_delete.py:1606]: Here the number is wrong.

File: tests/python_client/testcases/test_delete.py [hunk 4]
```diff
@@ -1064,3 +1068,579 @@ def test_delete_sealed_only(self):
...
 
         collection_w.query(expr, check_task=CheckTasks.check_query_empty)
+
+class TestDeleteString(TestcaseBase):
+    """
+    Test case of delete interface with string 
+    """
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_delete_entities_repeatedly_with_string(self):
```
Comment (binbinlv) [inline tests/python_client/testcases/test_delete.py:1078]: Do we have the case covering the scenario that delete using the same expression twice before query or search?

File: tests/python_client/testcases/test_delete.py [hunk 5]
```diff
@@ -1064,3 +1068,579 @@ def test_delete_sealed_only(self):
...
+        # delete same id 0 from default_partition, and query on it get empty result
+        collection_w.delete(default_string_expr, partition_name=ct.default_partition_name)
+        assert collection_w.num_entities == tmp_nb * 2
+        collection_w.load()
+        collection_w.query(default_string_expr, partition_names=[ct.default_partition_name],
+                           check_task=CheckTasks.check_query_empty)
+
+        # query on partition_w with id 0 and get an result
+        collection_w.query(default_string_expr, partition_names=[partition_w.name],
+                           check_task=CheckTasks.check_query_results, check_items={exp_res: query_string_tmp_expr})
```
Comment (binbinlv) [inline tests/python_client/testcases/test_delete.py:1296]: “query_string_tmp_expr” here should be modified.

File: tests/python_client/testcases/test_delete.py (hunk 4 above)
Comment (jingkl) [reply tests/python_client/testcases/test_delete.py:1079]: Not sure what the significance is

File: tests/python_client/testcases/test_delete.py (hunk 1 above)
Comment (jingkl) [reply tests/python_client/testcases/test_delete.py:1090]: No call to num_entity should correspond to the growing segment, after the call to the sealed

**Pull request merged**